### PR TITLE
Allow fallback to be a function

### PR DESCRIPTION
**Pull request by nzakas**
Intended to do the same thing as https://github.com/estools/estraverse/pull/68

### CODE REVIEW

**Comment by RReverser** [comment]
Sounds good to me.

File: esrecurse.js [hunk 1]
```diff
@@ -63,6 +63,12 @@
             ? assign({}, estraverse.VisitorKeys, options.childVisitorKeys)
             : estraverse.VisitorKeys;
         this.__fallback = options.fallback === 'iteration';
+        this.__fallbackFunction = objectKeys;
```
Comment (michaelficarra) [inline esrecurse.js]: I would make the same comment here about not having two pieces of state.

**Comment by nzakas** [comment]
Updated

**Comment by nzakas** [comment]
Since that was the only feedback, I'm going to merge so I can get these changes published.

**Pull request merged**